velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -854,5 +855,100 @@ TEST_F(Re2FunctionsTest, regexExtractAllBadArgs) {
   EXPECT_THROW(eval("123", "[a-z]+", 1), VeloxException);
 }
 
+TEST_F(Re2FunctionsTest, tryException) {
+  // Assert we throw without try.
+  VELOX_ASSERT_THROW(
+      evaluateOnce<std::string>(
+          "re2_extract(c0, c1)",
+          std::optional("123"),
+          std::optional("V)%(&r_b2o&Xw")),
+      "invalid regular expression");
+
+  auto patternVector = makeConstant("V)%(&r_b2o&Xw", 3);
+  auto oneGoodPatternVector =
+      makeNullableFlatVector<StringView>({"V)%(&r_b2o&Xw", ".*", std::nullopt});
+  auto stringVector = makeFlatVector<std::string>({"abc", "mno", "pqr"});
+  auto input = makeRowVector({stringVector, patternVector});
+  auto oneGoodInput = makeRowVector({stringVector, oneGoodPatternVector});
+
+  // Assert we can handle trys safely for re2_extract.
+  {
+    auto result =
+        evaluate<SimpleVector<StringView>>("try(re2_extract(c0, c1))", input);
+    assertEqualVectors(makeNullConstant(TypeKind::VARCHAR, 3), result);
+
+    // Atleast one non null result.
+    result = evaluate<SimpleVector<StringView>>(
+        "try(re2_extract(c0, c1))", oneGoodInput);
+    assertEqualVectors(
+        makeNullableFlatVector<StringView>({std::nullopt, "mno", std::nullopt}),
+        result);
+  }
+
+  // Try the same with re2match.
+  {
+    auto result = evaluate<SimpleVector<bool>>("try(re2_match(c0, c1))", input);
+    assertEqualVectors(makeNullConstant(TypeKind::BOOLEAN, 3), result);
+
+    // At least one non null result.
+    result =
+        evaluate<SimpleVector<bool>>("try(re2_match(c0, c1))", oneGoodInput);
+    assertEqualVectors(
+        makeNullableFlatVector<bool>({std::nullopt, true, std::nullopt}),
+        result);
+  }
+
+  // Try the same with re2search.
+  {
+    auto result =
+        evaluate<SimpleVector<bool>>("try(re2_search(c0, c1))", input);
+    assertEqualVectors(makeNullConstant(TypeKind::BOOLEAN, 3), result);
+
+    // At least one non null result.
+    result =
+        evaluate<SimpleVector<bool>>("try(re2_search(c0, c1))", oneGoodInput);
+    assertEqualVectors(
+        makeNullableFlatVector<bool>({std::nullopt, true, std::nullopt}),
+        result);
+  }
+
+  // Ensure like works well with Try.
```

Comment:
Would you also test the case when only some rows throw an error?

Reply:
Hi Masha, because like requires the pattern to be provided at build time , its not possible to supply a bad pattern for a particular row , as you can only supply a good pattern or a bad one and thus fail all rows or none. So atm I am not able to create test case for errors on particular rows unless I generate bad strings - which isnt that straightforward.